sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/PushReceiver.java

```diff
@@ -38,6 +42,13 @@ public void onReceive(Context context, Intent intent) {
         /* Received message action. */
         else if (INTENT_ACTION_RECEIVE.equals(action)) {
             Push.getInstance().onMessageReceived(context, intent);
+
+            /* Prevent handling message by not initialized firebase. */
+            if (!FirebaseUtils.isFirebaseAvailable()) {
+                AppCenterLog.warn(LOG_TAG, "Abort \"" + INTENT_ACTION_RECEIVE + "\" broadcast " +
+                        "because firebase is not available.");
+                abortBroadcast();
```

Comment:
Does not work on my Pixel device with Android P I still have the crash.

Another way is to call       

```
PackageManager pm = getPackageManager();
        pm.setComponentEnabledSetting(new ComponentName(this, "com.google.firebase.messaging.FirebaseMessagingService"),
                PackageManager.COMPONENT_ENABLED_STATE_DISABLED, PackageManager.DONT_KILL_APP);
```

say in `Push.registerPushToken`.

Since the state is persisted, we should enable again it if we detect firebase available in that method, so that it works when they migrate.